### ISSUE TITLE
Notifications not sent unless reportedItemFlags is set

### ISSUE THREAD

**Issue by homebysix** [lindegroup/autopkgr#533]
It looks like as recently as AutoPkgr 1.4.1, notifications are not sent unless the `reportedItemFlags` setting is present in the preferences. To reproduce, try this:

1. Set the schedule to run every 1 hour (so you don't have to wait long).
    <img width="329" alt="screen shot 2016-05-19 at 9 51 44 pm" src="https://cloud.githubusercontent.com/assets/7801391/15417687/6c986708-1e0d-11e6-8955-d867a5bc7f49.png">

2. Also make sure you're configured to receive email notifications.

3. Remove the com.github.autopkg.download.googlechrome folder from the AutoPkg cache.

4. Add the com.github.autopkg.download.googlechrome recipe to the active recipe list.

5. Quit AutoPkgr, then remove the `reportedItemFlags` setting from the preferences:

    ```
    $ defaults delete com.lindegroup.AutoPkgr reportedItemFlags
    ```

6. Relaunch AutoPkgr and observe that the notification trigger is set to "Report All Items." (Just observe, don't change it.)
    <img width="470" alt="screen shot 2016-05-19 at 9 51 27 pm" src="https://cloud.githubusercontent.com/assets/7801391/15417700/8491d696-1e0d-11e6-92db-e57a8806363a.png">

7. After the next scheduled run, you'll notice that __no email is received__.

8. Go into the notification trigger settings and toggle to "Report Some Items" and then back to "Report All Items."

9. Remove the com.github.autopkg.download.googlechrome folder from the AutoPkg cache again.

10. After the next scheduled run, you will receive an email notification as expected.

I think the best solution would be when AutoPkgr launches and there is no `reportedItemFlags` preference, it sets `reportedItemFlags = 1` by default.

I'm can't say for sure, but it seems reasonable that people who updated from AutoPkgr 1.3.2 to AutoPkgr 1.4+ without subsequently adjusting the notification trigger settings might not have a `reportedItemFlags` preference set.

@eahrold, could you try to reproduce this one and make sure it's not just me? I've been able to reproduce this on two Macs, both with 10.11.5 and AutoPkgr 1.4.1. It might be the issue at the core of that pesky #508.

**Comment by eahrold** [comment]
A HA!!

When launching the app as a GUI it gets set here https://github.com/lindegroup/autopkgr/blob/master/AutoPkgr/Views-Controllers-XIB/LGAppDelegate.m#L54-L55

However it won't show up doing a `defaults read` because it only exists in memory unless it's explicitly set.

But this is never called for background run... hence the issue.
I'll move it further back in the chain.

**Comment by homebysix** [comment]
Included in 1.4.2, just released.

**Comment by soberhofer** [comment]
I have found an issue that seems related to it so i'll just put it in here.

For me Notifications only work if reportedItemFlags is equal to 1.

If i edit the Flag via gui or defaults (even if i choose "Report all Items", which sets the Flag to 255) i get no notifications at all.

Only when i set the Flag to 1 via defaults i get notifications, which means i cannot set it to only notify me on new munki imports.

To test the notifications I have not run AutoPkgr manually. I used `sudo launchctl stop com.lindegroup.AutoPkgr.schedule && sudo launchctl start com.lindegroup.AutoPkgr.schedule` which emulates a scheduled run.

I use local Push notifications and a Slack Webhook. Since i also don't get local push notifications, i think this issue should not be related to slack.
